### PR TITLE
feat(generator): document retry policies

### PR DESCRIPTION
**Pull request by coryan**
Part of the work for #9434

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/12024)

### CODE REVIEW

**Comment by coryan** [comment]
@dbolduc there are some Bigtable changes here that you may want to review.

File: generator/internal/connection_generator.cc [hunk 1]
```diff
@@ -83,22 +83,124 @@ Status ConnectionGenerator::GenerateHeader() {
...
+    return impl_.IsPermanentFailure(status);
+  }
+  std::unique_ptr<$retry_policy_name$> clone() const override {
+    return std::make_unique<$limited_error_count_retry_policy_name$>(
+        maximum_failures());
+  }
+
+  // This is provided only for backwards compatibility.
+  using BaseType = $retry_policy_name$;
+ private:
```
Comment (devbww) [inline generator/internal/connection_generator.cc:138]: Nit: Insert separator line.

File: generator/internal/connection_generator.cc [hunk 2]
```diff
@@ -83,22 +83,124 @@ Status ConnectionGenerator::GenerateHeader() {
...
+    return impl_.IsPermanentFailure(status);
+  }
+  std::unique_ptr<$retry_policy_name$> clone() const override {
+    return std::make_unique<$limited_time_retry_policy_name$>(
+        maximum_duration());
+  }
+
+  // This is provided only for backwards compatibility.
+  using BaseType = $retry_policy_name$;
+ private:
```
Comment (devbww) [inline generator/internal/connection_generator.cc:200]: Nit: Insert separator line.

**Review by devbww — APPROVED** [review]

File: generator/internal/connection_generator.cc (hunk 2 above)
Comment (coryan) [reply generator/internal/connection_generator.cc:202]: Done

**Comment by dbolduc** [comment]
> @dbolduc there are some Bigtable changes here that you may want to review.

changes in `google/cloud/bigtable/...` LGTM

**Pull request merged**